config/config.go

```diff
@@ -81,6 +81,12 @@ type HealthCheckConfig struct {
 	HealthCheckValueTimeout int64 `json:"health_check_value_timeouts"`
 }
 
+type DnsCacheConfig struct {
+	Enabled       bool  `json:"enabled"`
+	TTL           int64 `json:"ttl"`
+	CheckInterval int64 `json:"check_interval"`
```

Comment:
Looks like this value regulates how often in-app cache will be wiping out expired items (for memory usage purposes). Why do we need to care about it on tyk.conf level?

Reply:
yes, I get that `CheckInterval` is added here in config because hard-coding doesn't allow it to be changed.
my question was why we want to have it configured. if you look up code by `cache.New` you will see that we use in-app caching a lot and none of those have cleanup interval as config value.
Customers will be asking why do they need to configure this, I don't see any reasonable answer tbh because it is something which controls in-app memory management and doesn't influence items expiration. That might introduce some noise in support.

Reply:
There's a doc for this value, which we can improve or add more information in order to explain the purpose of this parameter:
```
dns_cache.check_interval
Valid range: [-1;18446744073709551615]
Units: seconds
Default value: `0`
This setting allows to specify interval in seconds of expired records purging.
Setting this value to `-1` prevents checks from being executed and therefore dns cache records from being removed.
```

But in general, I don't have a clear vision regarding your remarks?As managing the inter

Reply:
There's a doc for this value, which we can improve or add more information in order to explain the purpose of this parameter:

```
dns_cache.check_interval
Valid range: [-1;18446744073709551615]
Units: seconds
Default value: `0`
This setting allows to specify interval in seconds of expired records purging.
Setting this value to `-1` prevents checks from being executed and therefore dns cache records from being removed.
```

But as exposing this value to configuration can be harmful, I've removed it from there and from docs. 
related commit - ed314a2894a5617e72fdd80971a45565dec91ec4